### DISCUSSION TITLE
I have an important question(s)

### DISCUSSION THREAD

**Discussion by Wjxfi** [NimaKhajehpour/MyMood#12] [Q&A]
will everything from the backup be saved, even if the backup is huge? If I write something for 20 years, can the application handle that amount of information? Can it work well with that amount of information? Can it restore that amount of information from the backup? will the app be able to create such a huge backup?

If I want to use this app all the time, I need to know that

undoubtedly my phone will break at some point. Then it will need to be transferred (backup)

**Comment by NimaKhajehpour** [comment]
Hey there @Wjxfi thanks for considering to use this app for a long time, as for the data backup it should be able to backup all the data and get all the data again after the import is done. generally how the app works is that it gets every single data that is in the databse (other than the setting for theme and other settings that may come in the future) and makes a XML file (as you know) and then for importing the data it will read the xml and populates the database. so theoretically it will do it as long as it can read the database, write it as an xml file and read the xml file and write the database.

**Comment by Wjxfi** [comment]
thanks for the answer😀
even if the xml file is large, will it still read quickly and almost without delay?
Some other applications have problems with this, that's why I ask

**Comment by NimaKhajehpour** [comment]
hopefully it will take close to no time to do it, but i am not sure at this moment, i might use some heavy dummy code for that and answer your question more accurately. also i opened discussions for this repo and app so feel free to use the discussions too :heart:
